### DISCUSSION TITLE
Add "delete file on deselection" relational trigger to image interface

### DISCUSSION THREAD

**Discussion by LasseRosenow** [directus/directus#12766] [Feature Requests]
Other interfaces o2m have the possibility to delete the items from the M collection if they are deselected on the O collection.

Especially for images this would be a major improvement, as it would help to keep the media library clean.

**Comment by rijkvanzanten** [comment]
Great idea!

The difficulty in implementing this this is that we currently rely on the database's foreign key constraints with a CASCADE to delete the related items. This works for the `o2m->junction row (jr)` part, but doesn't trigger on the `jr->file`, as that connection doesn't require any cascading behavior on the database 🤔 In order to support this, we'd have to re-implement the databases cascading magic

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
